tools/walletextension/storage/database/mariadb/mariadb.go

```diff
@@ -142,3 +144,25 @@ func (m *MariaDB) GetAllUsers() ([]common.UserDB, error) {
 
 	return users, nil
 }
+
+func (m *MariaDB) StoreTransaction(rawTx string, userID []byte) error {
+	stmt, err := m.db.Prepare("INSERT INTO transactions(user_id, tx_hash, tx) VALUES (?, ?, ?)")
+	if err != nil {
+		return err
+	}
+	defer stmt.Close()
+
+	// Calculate tx hash
+	rawTxBytes, err := hex.DecodeString(rawTx[2:])
+	if err != nil {
+		panic(err)
```

Comment:
We shouldn't panic here imo, just log the error and store it without the tx hash. (Same on the sqlite impl)

Reply:
fixed, good catch!